internal/collector/secret_refs.go

```diff
@@ -0,0 +1,171 @@
+package collector
```

Comment:
I think secret handling deserves its own package. And we should unify it will the LogPipeline controller.